scene/gui/text_edit_menu.h

```diff
@@ -0,0 +1,93 @@
+/**************************************************************************/
+/*  text_edit_menu.h                                                      */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEXT_EDIT_MENU_H
+#define TEXT_EDIT_MENU_H
+
+#include "scene/gui/popup_menu.h"
```

Comment:
Trying to forward declare it gave me errors in `audio_stream_import_settings.h` and `node_3d_editor_plugin.h` so I think it's best to leave it as it is for now.

Reply:
You need to restore the include in `LineEdit`, then you can safely forward declare

Reply:
Tried that, then added forward-declaration for TextDirection to text_edit_menu.h. Now I'm getting errors like: '_get_menu_action_accelerator': unknown override specifier

Reply:
Okay, found a way around it. Had to forward declare an enum, and then include `control.h` because I couldn't find a way to forward decleare a scoped enum (`TextDirection` specifically)